django/db/backends/base/operations.py

```diff
@@ -792,3 +792,31 @@ def prepare_join_on_clause(self, lhs_table, lhs_field, rhs_table, rhs_field):
     def format_debug_sql(self, sql):
         # Hook for backends (e.g. NoSQL) to customize formatting.
         return sqlparse.format(sql, reindent=True, keyword_case="upper")
+
+    def format_json_path_numeric_index(self, num):
+        """
+        Hook for backends to customize indexing in JSON paths.
+        """
+        return "[%s]" % num
+
+    def compile_json_path(self, key_transforms, include_root=True):
+        """
+        Hook for backends to customize all aspects of JSON path construction.
+        """
+        path = ["$"] if include_root else []
+        for key_transform in key_transforms:
+            try:
+                num = int(key_transform)
+            except ValueError:  # non-integer
+                path.append(".")
+                path.append(json.dumps(key_transform))
+            else:
+                if (
+                    num < 0
+                    and not self.connection.features.supports_json_negative_indexing
+                ):
+                    raise NotSupportedError(
+                        "negative indexing is not supported on this database backend."
```

Comment:
Nit:

```suggestion
                        "Negative indexing is not supported on this database backend."
```

or perhaps:


```suggestion
                        "Using negative JSON array indices is not supported on this database backend."
```